jigoshop-software.php

```diff
@@ -1414,16 +1453,119 @@ function ajax_jgs_lost_license() {
 			exit;
 		}
 
+
+		/**
+		 * Unsubscribe from activation notification emails.
+		 *
+		 * @since 2.7
+		 * @author Anton Iancu <anton.iancu@gmail.com>
+		 */
+
+		function ajax_jgs_activation_unsubscribe() {
+				//reset message and reponse vars
+				$message = null;
+				$response = null;
+				$unsubscribed = null;
+
+				//	 nonce verification
+				if ( $_POST['jgs_activation_unsubscribe_nonce'] && !wp_verify_nonce( $_POST['jgs_activation_unsubscribe_nonce'], 'jgs_activation_unsubscribe' ) ) $messages['nonce'] = __( 'An error has occurred, please try again', 'jigoshop-software' );
+
+				$license_key = sanitize_text_field( $_POST['jgs_license'] );
+				$email_address = sanitize_email( $_POST['jgs_email'] );
+
+				if ( empty ( $email_address ) || empty( $license_key ) ) {
+					$success = false;
+					$message = "Please enter your email address and license key";
+					return false;
+				}
+
+				$client_order = $this->is_valid_license_key( $license_key, $email_address, null, null, false, false, true ); //will return the order id and data in an array.
+				$data = $client_order['order_data'];
+				$order_id = $client_order['order_id'];
+				$data['activation_email_optout'] = "on"; //Activate the optout option for the order
+				$unsubcribed = update_post_meta( $order_id, 'order_data', $data );//Update the options in the db
+
+				if ( $unsubcribed ) {
+					$success = true;
+					$message = "OK, you will no longer receive activation emails.";
+				} else {
+					$success = false;
+					$message = "This email address and license combination could not be found. Please verify your license key and email used to purchase it.";
+				}
+				header( 'Content-Type: application/json' );
```

Comment:
you should use `wp_send_json_success()` https://codex.wordpress.org/Function_Reference/wp_send_json_success or `_error()` https://codex.wordpress.org/Function_Reference/wp_send_json_error